git/gogit/client.go

```diff
@@ -341,6 +341,12 @@ func (g *Client) Commit(info git.Commit, commitOpts ...repository.CommitOption)
 	}
 
 	if options.Signer != nil {
+		if options.SignerPassphrase != "" {
```

Comment:
Instead of doing this, and thereby potentially keeping the passphrase around in memory for an extensive period of time. Wouldn't it simply be possible to run the `Decrypt` operation on the private key before providing it as a signer to options?

Reply:
Yeah but I fail to see how that's an issue? The passphrases are not going to be more than a few bytes, so it's affect on memory consumption should be negligible. Also, a primary reason for this is to make it easier for the consumers of this package. Both flux2 and IAC need this logic, so this can be used in multiple places by us too.

Reply:
okay closing this PR for now then